driver/src/logging.rs

```diff
@@ -21,47 +27,64 @@ pub fn init(filter: impl AsRef<str>) -> (Logger, GlobalLoggerGuard) {
     (logger, guard)
 }
 
-pub struct CustomFormatter<D: Decorator> {
-    decorator: D,
+/// Uses one decorator for `Error` and `Critical` log messages and the other for
+/// the rest.
+pub struct CustomFormatter<T0, T1> {
+    err_decorator: T0,
+    rest_decorator: T1,
 }
 
-impl<D: Decorator> CustomFormatter<D> {
-    fn new(decorator: D) -> Self {
-        Self { decorator }
+impl<T0, T1> CustomFormatter<T0, T1> {
+    fn new(err_decorator: T0, rest_decorator: T1) -> Self {
+        Self {
+            err_decorator,
+            rest_decorator,
+        }
     }
 }
 
-impl<D: Decorator> Drain for CustomFormatter<D> {
+impl<T0: Decorator, T1: Decorator> Drain for CustomFormatter<T0, T1> {
     type Ok = ();
     type Err = std::io::Error;
     fn log(
         &self,
         record: &Record,
         values: &OwnedKVList,
     ) -> std::result::Result<Self::Ok, Self::Err> {
-        self.decorator.with_record(record, values, |mut decorator| {
-            decorator.start_timestamp()?;
-            slog_term::timestamp_utc(&mut decorator)?;
+        match record.level() {
+            Level::Error | Level::Critical => log_to_decorator(&self.err_decorator, record, values),
```

Comment:
can we include warnings here? It caught an issue in #615

Reply:
I don't think we should. The way I imagined it was that anything on stderr is severe enough for us to get an alert on Slack. But warnings are not that severe.

Reply:
But then we would have merged #615 with the issue. I’m ok with CI being more restrictive than our alert system.